clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -109,22 +116,16 @@ struct BuiltinTypeDeclBuilder {
   }
 
   BuiltinTypeDeclBuilder &
-  addHandleMember(Sema &S, ResourceClass RC, ResourceKind RK, bool IsROV,
-                  bool RawBuffer,
+  addHandleMember(ResourceClass RC, ResourceKind RK, bool IsROV, bool RawBuffer,
                   AccessSpecifier Access = AccessSpecifier::AS_private) {
-    if (Record->isCompleteDefinition())
-      return *this;
+    assert(!Record->isCompleteDefinition() && "record is already complete");
 
     ASTContext &Ctx = S.getASTContext();
     TypeSourceInfo *ElementTypeInfo = nullptr;
 
     QualType ElemTy = Ctx.Char8Ty;
```

Comment:
Pre-existing side-note: Does this mean that `ByteAddressBuffer` will have an element type of `i8`?  Shouldn't it be `void`, to clearly differentiate this case?

Reply:
`i8` seems like a good default to me. We can decide if we want to use `void` instead when we implement the `ByteAddressBuffer`.